addons/website/models/ir_qweb.py

```diff
@@ -36,16 +37,33 @@ def _get_template_cache_keys(self):
         """ Return the list of context keys to use for caching ``_compile``. """
         return super()._get_template_cache_keys() + ['website_id']
 
+    def _check_user_can_modify(self, record):
+        """ Verify that the current user can modify the given record.
+
+        :param record: record on which to perform the check
+        :raise AccessError: if the operation is forbidden
+        """
+        record.check_access_rights('write')
+        record.check_access_rule('write')
+
     def _prepare_frontend_environment(self, values):
         """ Update the values and context with website specific value
             (required to render website layout template)
         """
         irQweb = super()._prepare_frontend_environment(values)
 
         current_website = request.website
+        editable = irQweb.env.user.has_group('website.group_website_designer')
         has_group_restricted_editor = irQweb.env.user.has_group('website.group_website_restricted_editor')
-        editable = has_group_restricted_editor
-        translatable = editable and irQweb.env.context.get('lang') != irQweb.env['ir.http']._get_default_lang().code
+        translatable = has_group_restricted_editor and irQweb.env.context.get('lang') != irQweb.env['ir.http']._get_default_lang().code
+        if translatable and not editable and 'main_object' in values and has_group_restricted_editor:
```

Comment:
```suggestion
        if translatable and not editable and 'main_object' in values:
```

` and has_group_restricted_editor`is already implicit by ìf translatable`

Reply:
Moved stuff around. Now performing check only when trying to enter translate mode.
I will now adapt the test.